src/sql/src/plan/statement/ddl.rs

```diff
@@ -1093,6 +1120,45 @@ fn get_encoding_inner<T: sql_parser::ast::AstInfo>(
     }))
 }
 
+fn compile_proto(schema: &str) -> Result<Vec<u8>, anyhow::Error> {
+    // Write schema string to file to compile it.
+    let schema_bytes = strconv::parse_bytes(schema)?;
+    let include_dir = tempfile::tempdir()?.into_path();
+    let mut file = tempfile::NamedTempFile::new_in(&include_dir)?;
+    file.write_all(&schema_bytes)?;
+    file.flush()?;
+
+    // Destroy and recreate the build directory, in case any inputs have
+    // been deleted since the last invocation.
+    let out_dir = tempfile::tempdir()?.into_path();
+    match fs::remove_dir_all(&out_dir) {
+        Ok(()) => (),
+        Err(e) if e.kind() == io::ErrorKind::NotFound => (),
+        Err(e) => {
+            return Err(anyhow::Error::new(e)).with_context(|| {
+                format!("removing existing out directory {}", out_dir.display())
+            })?
+        }
+    }
+    fs::create_dir(&out_dir)
+        .with_context(|| format!("creating out directory {}", out_dir.display()))?;
+
+    // Compile schema string.
+    protoc::Protoc::new()
+        .include(&include_dir)
+        .input(file.path())
+        .compile_into(&out_dir)?;
+
+    for dir_result in fs::read_dir(&out_dir)? {
+        let dir_entry = dir_result?;
+        if dir_entry.file_name() == "file_descriptor_set.pb" {
+            return Ok(fs::read(&dir_entry.path())?);
+        }
+    }
+
+    return Err(anyhow::anyhow!("missing compiled proto"));
```

Comment:
this seems like it could be

```suggestion
    Ok(fs::read(out_dir.join("file_descriptor_set.pb").path())?)
```